migrations/20230523170000-create-collective-orders-stats.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`
+      CREATE MATERIALIZED VIEW IF NOT EXISTS "CollectiveOrderStats" AS
+      WITH ma AS (
+        SELECT
+          o."CollectiveId",
+          DATE_TRUNC('month', o."createdAt") AS "month",
+          ROUND(COALESCE(AVG(CASE WHEN o."status" = 'ERROR' THEN 1 ELSE 0 END), 0), 5)::FLOAT AS "errorRate",
+          ROUND(COALESCE(AVG(CASE WHEN o."status" IN ('PAID', 'ACTIVE') THEN 1 ELSE 0 END), 0), 5)::FLOAT AS "successRate",
+          COUNT(*) AS "numberOfOrders"
+        FROM
+            "Orders" o
+        LEFT JOIN "Collectives" c ON c."id" = o."CollectiveId"
+        WHERE
+          o."deletedAt" IS NULL
+          AND o."createdAt" > NOW() - INTERVAL '1 year'
+          AND c."deletedAt" IS NULL
+          AND c."isActive" IS TRUE
+        GROUP BY
+          o."CollectiveId",
+          "month"
+        ORDER BY "month" ASC
+      ), acc AS (
+        SELECT
+          "CollectiveId",
+          "month",
+          RANK() OVER w AS "months",
+          "errorRate",
+          AVG("errorRate") OVER w AS "accErrorRate",
+          "successRate",
+          AVG("successRate") OVER w AS "accSuccessRate",
+          SUM("numberOfOrders") OVER w AS "accNumberOfOrders"
+        FROM ma
+        WINDOW w AS (PARTITION BY "CollectiveId" ORDER BY "month" DESC)
+      )
+      SELECT
+        "CollectiveId",
+        SUM("accErrorRate") FILTER (WHERE "months" = 1) AS "errorRate1M",
+        SUM("accErrorRate") FILTER (WHERE "months" = 3) AS "errorRate3M",
```

Comment:
Shouldn't that be `<=`? Otherwise, we're going to sum only the data from 3 months ago instead of `SUM(month,month2,month3)`. Same below, I'm surprised by these `=`, but maybe I'm missing something.

```suggestion
        SUM("accErrorRate") FILTER (WHERE "months" <= 3) AS "errorRate3M",
```

Reply:
The accumulated sum is being done in the rolling window above. Here, there will be only a single value per month per collective and I'm using SUM just as an accumulator for grouping.

Reply:
So `month = 3` contains the aggregated info for the last 3 months, not just the info for `currentMonth - 3`?

Reply:
Exactly.

Reply:
It is aggregating with:
```sql
SELECT
  "CollectiveId",
  "month",
  RANK() OVER w AS "months",
  "errorRate",
  AVG("errorRate") OVER w AS "accErrorRate",
  "successRate",
  AVG("successRate") OVER w AS "accSuccessRate",
  SUM("numberOfOrders") OVER w AS "accNumberOfOrders"
FROM ma
WINDOW w AS (PARTITION BY "CollectiveId" ORDER BY "month" DESC)
```

This means month 3 has the average of the last 3 months.